servatrice/src/serversocketinterface.cpp

```diff
@@ -747,6 +749,45 @@ Response::ResponseCode ServerSocketInterface::cmdReplayDeleteMatch(const Command
 // MODERATOR FUNCTIONS.
 // May be called by admins and moderators. Permission is checked by the calling function.
 
+Response::ResponseCode ServerSocketInterface::cmdGetBanHistory(const Command_GetBanHistory &cmd, ResponseContainer &rc)
+{
+    if (!sqlInterface->checkSql())
+        return Response::RespInternalError;
+
+    QString userName = QString::fromStdString(cmd.user_name());
+
+    //lookup users ban history from database
+    QSqlQuery *query = sqlInterface->prepareQuery("SELECT A.id_admin, A.time_from, A.minutes, A.reason, A.visible_reason, B.name AS name_admin FROM {prefix}_bans A LEFT JOIN {prefix}_users B ON A.id_admin=B.id WHERE A.user_name = :user_name");
+    query->bindValue(":user_name", userName);
+
+    if (!sqlInterface->execSqlQuery(query))
+        return Response::RespInternalError;
+
+    //build response container information from results
+    QString adminID,adminName,banTime,banLength,banReason,visibleReason;
+    Response_BanHistory *re = new Response_BanHistory;
+    while (query->next()){
+
+        adminID = query->value(0).toString();
+        banTime = query->value(1).toString();
+        banLength = query->value(2).toString();
+        banReason = query->value(3).toString();
+        visibleReason = query->value(4).toString();
+        adminName = query->value(5).toString();
+
+        re->add_admin_id(adminID.toStdString());
+        re->add_admin_name(adminName.toStdString());
+        re->add_ban_time(banTime.toStdString());
+        re->add_ban_length(banLength.toStdString());
+        re->add_ban_reason(banReason.toStdString());
+        re->add_visible_reason(visibleReason.toStdString());
```

Comment:
Even if you made the actual response repeatable rather than each item this behavior wouldn't change.  You still need to add repeated response information.